earthfile2llb/converter.go

```diff
@@ -1143,6 +1181,23 @@ func (c *Converter) Import(ctx context.Context, importStr, as string, isGlobal,
 	return c.varCollection.Imports().Add(importStr, as, isGlobal, currentlyPrivileged, allowPrivilegedFlag)
 }
 
+// Cache handles a `CACHE` command in a Target.
+// It appends run options to the Converter which will mount a cache volume in each successive `RUN` command,
+// and configures the `Converter` to persist the cache in the image at the end of the target.
+func (c *Converter) Cache(ctx context.Context, path string) error {
```

Comment:
Should any new `FROM` encountered in the same target reset any cache dirs collected so far?

Reply:
as-is currently, it seems to lose the cached files after another `FROM`. 

I think it's probably safest in general to not try and keep the files (e.g. if they went from linux to windows, could those files have unexpected issues?)

Open to your opinion here on if we should do anything more. Possibilities include:
* Try to support allowing multiple FROMs (don't like it, probably buggy and high effort for unlikely use case)
* Restrict it (throw an error if there's another FROM after cache)
* Better handle it: drop cache dirs etc since they're not used, being a bit more efficient. maybe whats going on so user understands
* Just leave it as-is for now, not worth effort for unlikely edge case on a new feature we don't the adoption rate of yet

Reply:
I'd go with this option - it's probably what users would expect (`FROM` anyway resets everything else):

> Better handle it: drop cache dirs explicitly since they're not used (being a bit more efficient). maybe log what's going on so the user understands.

If someone wants to transfer the cache from one `FROM` to another, they should probably better use a different target.

Reply:
Wondering though if Warning log is the right way to communicate this, as it looks more like a serious error in red text than perhaps it is. 

Is there a more appropriate way to print something at more of an info-level within the converter logic? Just `fmt.Println`?

Reply:
I would maybe not log it at all?.. we don't really have a good way of logging things from the converter.

Reply:
removed the log